app/components/document-creator.js

```diff
@@ -0,0 +1,29 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+    tagName: 'document-creator',
+    classNames: 'document-creator',
+    documentTitle: null,
+    isCreating: false,
+
+    actions: {
+      createDocument: function () {
+        this.createDocument();
+      },
+    },
+
+    createDocument: function () {
+      var title = this.get('documentTitle').trim(),
+        component = this;
+
+      this.set('isCreating', true);
+
+      console.log('creating document...');
+      return this.get('documentSource').createManually(title).then(function (doc) {
+        component.sendAction('documentCreated', doc);
+        component.set('isCreating', false);
+      }, function (error) {
+        component.set('isCreating', false);
+      });
+    }
+});
```

Comment:
The component is relatively simple. There's one action and one property (`isCreating`), which we can use to indicate something's going on in the interface, Document creation takes a while, so the user needs a UI response. Otherwise, there's a good chance they'd click several times on the create button.